### ISSUE TITLE
callOnTapGesture not updating @State value?

### ISSUE THREAD

**Issue by grantneufeld** [nalexn/ViewInspector#185]
EDIT: **Nevermind.** I missed the part in the guide on needing to add special code to the view for cases like this. Oops.

---

I’m using `callOnTapGesture` to trigger an `onTapGesture` closure. Stepping through in the debugger (when running the unit test), I can see that the closure does get called, but the @State variable it updates (by calling `self.tapped.toggle()`, in my sample code below) does not actually get updated. (Running the app, however, does trigger the expected @State change in the view.)

My View:

```swift
import SwiftUI

struct InspectorExampleView: View {
    @State var tapped = false
    var body: some View {
        VStack {
            Text("Example")
                .onTapGesture {
                    self.tapped.toggle()
                }
            Text(tapped ? "Activated" : "Inactive")
        }
    }
}
```

My unit test:

```swift
@testable import MyApp
import ViewInspector
import XCTest

class InspectorExampleViewTests: XCTestCase {
    func testExampleText_whenTapped_changesBool() throws {
        let sutView: InspectorExampleView! = InspectorExampleView()
        let sut = try sutView.inspect().view(InspectorExampleView.self)
        let example = try sut.find(text: "Example")
        XCTAssertNoThrow(try sut.find(text: "Inactive")) // ✅ test passes
        try example.callOnTapGesture()
        XCTAssertNoThrow(try sut.find(text: "Activated")) // 🛑 test fails
        XCTAssertThrowsError(try sut.find(text: "Inactive")) // 🛑 test fails
    }
}

extension InspectorExampleView: Inspectable {
}

```

**Comment by grantneufeld** [comment]
Oops. I missed the part in the guide doc that shows how we need to add special code to the view, and write the tests a bit differently, to handle stuff like this.

**Comment by aaryankotharii** [comment]
Hi @grantneufeld, can you please the what special code we need to add to test this ?
